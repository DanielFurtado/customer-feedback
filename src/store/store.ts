import { configureStore } from '@reduxjs/toolkit';
import ratingReducer from './rating/rating.reducer';

export const store = configureStore({
  reducer: {
    feedbackList: ratingReducer
  }
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
