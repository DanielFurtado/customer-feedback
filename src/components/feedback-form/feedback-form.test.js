import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../store/store';
import { BrowserRouter as Router } from 'react-router-dom';
import FeedbackForm from './feedback-form';

describe(('Feedback Form'), () => {  
  const renderFeedbackForm = () => {
    return render(
      <Provider store={store}>
        <Router>
          <FeedbackForm />
        </Router>
      </Provider>
    );
  };

  it('renders', () => {
    let component = renderer.create(
      <Provider store={store}>
        <Router>
          <FeedbackForm />
        </Router>
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the basic fields', () => {
    renderFeedbackForm();

    expect(screen.getByRole('textbox', { 
      name: 'Name' 
    })).toBeInTheDocument();

    expect(screen.getByRole('textbox', { 
      name: 'Email' 
    })).toBeInTheDocument();

    expect(screen.getByRole('spinbutton', { 
      name: 'Rating' 
    })).toBeInTheDocument();

    expect(screen.getByRole('textbox', { 
      name: 'Comment' 
    })).toBeInTheDocument();

    expect(screen.getByRole('button', { 
      name: 'Submit' 
    })).toBeInTheDocument();
  });

  it('shows error messages', async () => {
    renderFeedbackForm();

    fireEvent.input(screen.getByRole('textbox', { name: 'Email' }), {
      target: { value: 'test wrong email' }
    });

    fireEvent.click(screen.getByText(/submit/i));
    
    await waitFor(() => {
      expect(screen.getByText('Please enter a comment.')).toBeInTheDocument();
    }); 
    expect(screen.getByText('Entered value does not match email format.')).toBeInTheDocument();
    expect(screen.getByText('Please enter a comment.')).toBeInTheDocument();
    expect(screen.getByText('Please select a rating')).toBeInTheDocument();
  });
});
