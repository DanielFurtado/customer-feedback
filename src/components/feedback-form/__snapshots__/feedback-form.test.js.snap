// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback Form renders 1`] = `
<form
  className="sc-bcXHqe iozQoF"
  noValidate={true}
>
  <h2>
    Feedback Form
  </h2>
  <div
    className="sc-gswNZR kDxzyM"
  >
    <div
      className="sc-dkrFOg jrznKg"
    >
      <div
        className="Form--group"
      >
        <label
          htmlFor="name"
        >
          Name *
        </label>
        <input
          aria-invalid="false"
          aria-label="Name"
          id="name"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Name"
          type="text"
        />
      </div>
      <div
        className="Form--group"
      >
        <label
          htmlFor="email"
        >
          Email *
        </label>
        <input
          aria-invalid="false"
          aria-label="Email"
          id="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Email"
          type="text"
        />
      </div>
      <div
        className="Form--group"
      >
        <label
          htmlFor="rating"
        >
          Rating *
        </label>
        <input
          aria-invalid="false"
          aria-label="Rating"
          id="rating"
          max="5"
          min="1"
          name="rating"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Rating (1-5 stars)"
          type="number"
        />
      </div>
    </div>
    <div
      className="sc-dkrFOg jrznKg"
    >
      <div
        className="sc-hLBbgP cqTXbJ"
      >
        <label
          htmlFor="comment"
        >
          Comment *
        </label>
        <textarea
          aria-invalid="false"
          aria-label="Comment"
          id="comment"
          name="comment"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Comment"
        />
      </div>
    </div>
  </div>
  <div
    className="sc-eDvSVe cMlANh"
  >
    <button
      className="sc-ftTHYK iynfzl"
      onClick={[Function]}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;
